go/enclave/enclave.go

```diff
@@ -277,29 +276,9 @@ func (e *enclaveImpl) Nonce(address gethcommon.Address) uint64 {
 	return s.GetNonce(address)
 }
 
-func (e *enclaveImpl) GetTransaction(txHash gethcommon.Hash) *common.L2Tx {
-	// todo - use the metadata stored in the database
-	hs := e.storage.FetchHeadState()
-	if hs == nil {
-		panic("should not happen")
-	}
-	rollup, found := e.storage.FetchRollup(hs.HeadRollup)
-	if !found {
-		log.Panic("could not fetch block's head rollup")
-	}
-
-	for {
-		txs := rollup.Transactions
-		for _, tx := range txs {
-			if bytes.Equal(tx.Hash().Bytes(), txHash.Bytes()) {
-				return tx
-			}
-		}
-		rollup = e.storage.ParentRollup(rollup)
-		if rollup == nil || rollup.Header.Number.Uint64() == common.L2GenesisHeight {
-			return nil
-		}
-	}
+func (e *enclaveImpl) GetTransaction(txHash gethcommon.Hash) (*common.L2Tx, gethcommon.Hash, uint64, uint64, error) {
+	// todo - joel - encrypt
+	return e.storage.GetTransaction(txHash)
```

Comment:
I had raised a ticket for this.
 I assigned you just now: https://github.com/obscuronet/obscuro-internal/issues/568

Reply:
Thanks